### PR TITLE
SDK is non static and logging/tracing controls are moved to the connectors

### PR DESCRIPTION
**Pull request by adamhathcock**
Needs https://github.com/specklesystems/speckle-sharp-sdk/pull/116 as a NuGet to fix up references and Solution

https://www.figma.com/board/h0HKFlItQRmI8WYQ5AGmbm/Connector-Dependencies?node-id=0-1&node-type=canvas&t=LRBQV8i4iEqGhBFM-0

### CODE REVIEW

File: Sdk/Speckle.Connectors.Utils/Connector.cs [hunk 1]
```diff
@@ -18,42 +23,33 @@ public static class Connector
...
+      VersionString,
+      Slug,
+      Assembly.GetExecutingAssembly().GetVersion(),
       new(
-        Console: false,
-        Otel: new(
-          Endpoint: "https://seq-dev.speckle.systems/ingest/otlp/v1/traces",
-          Headers: new() { { "X-Seq-ApiKey", "y5YnBp12ZE1Czh4tzZWn" } }
-        )
+        new SpeckleLogging(Console: true, Otel: null, MinimumLevel: SpeckleLogLevel.Warning),
```
Comment (JR-Morgan) [inline Sdk/Speckle.Connectors.Utils/Connector.cs:42]: Not logging Information logs?

Comment (JR-Morgan) [reply Sdk/Speckle.Connectors.Utils/Connector.cs:42]: Otel: null?

File: Sdk/Speckle.Connectors.Utils/SpeckleLogProvider.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿using Microsoft.Extensions.Logging;
+using Speckle.Connectors.Logging;
+
+namespace Speckle.Connectors.Utils;
+
+public sealed class SpeckleLogProvider(Logger speckleLogger) : ILoggerProvider
+{
+  public void Dispose() { }
+
+  public ILogger CreateLogger(string categoryName) => new SpeckleLogger(speckleLogger);
```
Comment (AlanRynne) [inline Sdk/Speckle.Connectors.Utils/SpeckleLogProvider.cs:10]: Are we not using the categoryName for anything? if so... should we?

**Review by AlanRynne — COMMENTED** [review]
Just a bunch of comments

File: Sdk/Speckle.Connectors.Utils/Connector.cs (hunk 1 above)
Comment (adamhathcock) [reply Sdk/Speckle.Connectors.Utils/Connector.cs:42]: by default, currently, this is not logging anywhere yet

File: Connectors/Autocad/Speckle.Connectors.AutocadShared/Plugin/AutocadRibbon.cs [hunk 3]
```diff
@@ -178,18 +177,9 @@ private bool TryLoadPngImgSource(string sourceName, out System.Windows.Media.Ima
...
-    {
-      SpeckleLog.Logger.Error(fnfEx, "Could not find app image source: {exceptionMessage}");
-    }
-    catch (NotImplementedException niEx)
-    {
-      SpeckleLog.Logger.Error(niEx, "App image source could not be loaded: {exceptionMessage}");
-    }
+    catch (FileLoadException) { }
+    catch (FileNotFoundException) { }
+    catch (NotImplementedException) { }
```
[the remark this answers is not in the document]
Comment (adamhathcock) [reply Connectors/Autocad/Speckle.Connectors.AutocadShared/Plugin/AutocadRibbon.cs:182]: there's a DI limitation that isn't straightforward to resolve

File: Sdk/Speckle.Connectors.Utils/Common/AssemblyExtensions.cs [hunk 4]
```diff
@@ -6,23 +6,23 @@ public static class AssemblyExtensions
...
-        assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion ?? "No version";
-      if (!string.IsNullOrEmpty(informationalVersion))
-      {
-        return informationalVersion;
-      }
-    }
-#pragma warning disable CA1031
-    catch (Exception)
-#pragma warning restore CA1031
+    // MinVer https://github.com/adamralph/minver?tab=readme-ov-file#version-numbers
```
[the remark this answers is not in the document]
Comment (adamhathcock) [reply Sdk/Speckle.Connectors.Utils/Common/AssemblyExtensions.cs:9]: This has been fixed

File: Sdk/Speckle.Connectors.Utils/SpeckleLogProvider.cs (hunk 2 above)
Comment (adamhathcock) [reply Sdk/Speckle.Connectors.Utils/SpeckleLogProvider.cs:10]: By default, it uses the class name when using ILogger<T>

File: Connectors/ArcGIS/Speckle.Connectors.ArcGIS3/SpeckleModule.cs [hunk 5]
```diff
@@ -32,7 +32,7 @@ public SpeckleModule()
 
     var builder = SpeckleContainerBuilder.CreateInstance();
     // init DI
-    _disposableLogger = Connector.Initialize(HostApplications.ArcGIS, GetVersion());
+    _disposableLogger = Connector.Initialize(HostApplications.ArcGIS, GetVersion(), builder);
```
[the remark this answers is not in the document]
Comment (adamhathcock) [reply Connectors/ArcGIS/Speckle.Connectors.ArcGIS3/SpeckleModule.cs:35]: I think this is temporary and will be resolved when fixing DI in general on the connectors project

File: Connectors/Autocad/Speckle.Connectors.AutocadShared/Plugin/AutocadRibbon.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (AlanRynne) [reply Connectors/Autocad/Speckle.Connectors.AutocadShared/Plugin/AutocadRibbon.cs:182]: Ok, I created an issue to have this tracked separately then:

https://linear.app/speckle/issue/CNX-509/autocad-ribbon-investigate-how-to-log-swallowed-exceptions-at-this

**Review by AlanRynne — APPROVED** [review]

**Pull request merged**